scripts/check_change_log/main.py

```diff
@@ -0,0 +1,109 @@
+# coding=utf-8
+import glob
+import os
+import subprocess as sp
+import time
+from pypi_tools.pypi import PyPIClient
+from pathlib import Path
+
+
+def find_report_name(result):
+    pattern = 'written to'
+    merged = 'merged_report'
+    for line in result:
+        idx = line.find(pattern)
+        idx1 = line.find(merged)
+        if idx > 0 and idx1 > 0:
+            return "_/" + line[idx + len(pattern):].replace(" ", "")
+
+    for line in result:
+        idx = line.find(pattern)
+        if idx > 0:
+            return "_/" + line[idx + len(pattern):].replace(" ", "")
+
+    return ''
+
+def create_foldor(name):
+    if not os.path.exists(name):
+        os.mkdir(name)
+        print("create successful")
+    else:
+        print("foldor has been created")
+
+def write_txt(foldor, text_name, text, version1, version2):
+    path = foldor + f"\{text_name}" + " " + version1 + "-" + version2 + r".txt"
+    with open(file=path, mode="w", encoding="utf-8") as file:
+
+        file.write(text)
+    print("txt create successful")
+
+def create_code_report(cmd):
+    info = sp.Popen(cmd,
+                    stderr=sp.STDOUT,
+                    stdout=sp.PIPE,
+                    universal_newlines=True,
+                    cwd=None,
+                    shell=False,
+                    env=None,
+                    encoding="utf-8",
+                    )
+
+    output_buffer = []
+    for line in info.stdout:
+        output_buffer.append(line.rstrip())
+    info.wait()
+    info_output = "\n".join(output_buffer)
+    return info_output.split('\n')
+
+if __name__ == '__main__':
+    # get sdk path
+    env = Path.cwd()
+    docker_path = env.parent.parent
+
+    # create docker env in sdk path
+    sp.call(fr"docker create -it --rm -h Change_log --name Change_log -v {docker_path}:/_ l601306339/autorest")
+    sp.call("docker start Change_log")
+
+    # install azure tools
+    sp.call(f'docker exec -it Change_log /bin/bash -c  "python _/scripts/dev_setup.py -p azure-core"  ')
+
+    # get all azure-mgmt-package paths
+    in_files = glob.glob(f'{docker_path}\sdk\*\*mgmt*')
```

Comment:
we need to be compatible with Linux. try:
`glob.glob(str(Path(f'{docker_path}/sdk/*/azure-mgmt-*')))`